src/state/queries/video/video-upload.web.ts

```diff
@@ -22,53 +22,56 @@ export const useUploadVideoMutation = ({
 
   return useMutation({
     mutationFn: async (video: CompressedVideo) => {
-      const uri = createVideoEndpointUrl('/upload', {
+      const uri = createVideoEndpointUrl('/xrpc/app.bsky.video.uploadVideo', {
         did: currentAccount!.did,
-        name: `${nanoid(12)}.mp4`, // @TODO what are we limiting this to?
+        name: `${nanoid(12)}.mp4`, // @TODO: make sure it's always mp4'
       })
 
       // a logged-in agent should have this set, but we'll check just in case
       if (!agent.pdsUrl) {
         throw new Error('Agent does not have a PDS URL')
       }
 
-      const {data: serviceAuth} =
-        await agent.api.com.atproto.server.getServiceAuth({
+      const {data: serviceAuth} = await agent.com.atproto.server.getServiceAuth(
+        {
           aud: `did:web:${agent.pdsUrl.hostname}`,
           lxm: 'com.atproto.repo.uploadBlob',
-        })
+        },
+      )
 
       const bytes = await fetch(video.uri).then(res => res.arrayBuffer())
 
       const xhr = new XMLHttpRequest()
-      const res = (await new Promise((resolve, reject) => {
-        xhr.upload.addEventListener('progress', e => {
-          const progress = e.loaded / e.total
-          setProgress(progress)
-        })
-        xhr.onloadend = () => {
-          if (xhr.readyState === 4) {
-            const uploadRes = JSON.parse(
-              xhr.responseText,
-            ) as UploadVideoResponse
-            resolve(uploadRes)
-            onSuccess(uploadRes)
-          } else {
+      const res = await new Promise<AppBskyVideoDefs.JobStatus>(
+        (resolve, reject) => {
+          xhr.upload.addEventListener('progress', e => {
+            const progress = e.loaded / e.total
+            setProgress(progress)
+          })
+          xhr.onloadend = () => {
+            if (xhr.readyState === 4) {
+              const uploadRes = JSON.parse(
+                xhr.responseText,
+              ) as AppBskyVideoDefs.JobStatus
+              resolve(uploadRes)
+              onSuccess(uploadRes)
+            } else {
+              reject()
+              onError(new Error('Failed to upload video'))
+            }
+          }
+          xhr.onerror = () => {
             reject()
             onError(new Error('Failed to upload video'))
           }
-        }
-        xhr.onerror = () => {
-          reject()
-          onError(new Error('Failed to upload video'))
-        }
-        xhr.open('POST', uri)
-        xhr.setRequestHeader('Content-Type', 'video/mp4') // @TODO how we we set the proper content type?
-        // @TODO remove this header for prod
-        xhr.setRequestHeader('dev-key', UPLOAD_HEADER)
-        xhr.setRequestHeader('Authorization', `Bearer ${serviceAuth.token}`)
-        xhr.send(bytes)
-      })) as UploadVideoResponse
+          xhr.open('POST', uri)
+          xhr.setRequestHeader('Content-Type', 'video/mp4')
+          // @TODO remove this header for prod
```

Comment:
fine to leave this here, maybe could make it `@TODO VIDEO` or something so its easy to find